Example/ViewController+StyleManager.swift

```diff
@@ -0,0 +1,38 @@
+import UIKit
+import CoreLocation
+import MapboxNavigation
+import MapboxMaps
+
+extension ViewController : StyleManagerDelegate {
+    func location(for styleManager: MapboxNavigation.StyleManager) -> CLLocation? {
+        if let location = navigationMapView.mapView.location.latestLocation?.location {
+            return location
+        } else if let location = CLLocationManager.init().location {
+            return location
+        }
+        return nil
+    }
+    
+    func styleManager(_ styleManager: MapboxNavigation.StyleManager, didApply style: MapboxNavigation.Style) {
+        updateMapStyle(style)
+    }
+    
+    func updateMapStyle(_ style: MapboxNavigation.Style) {
+        if navigationMapView?.mapView.mapboxMap.style.uri?.rawValue != style.mapStyleURL.absoluteString {
+            navigationMapView?.mapView.mapboxMap.style.uri = StyleURI(url: style.mapStyleURL)
+        }
+        updateFeedbackViewControllerStyle()
+    }
+    
+    func updateFeedbackViewControllerStyle() {
+        if let feedbackViewController = presentedViewController as? FeedbackViewController {
+            for window in UIApplication.shared.windows {
+                if !window.isKind(of: NSClassFromString("UITextEffectsWindow") ?? NSString.classForCoder()),
+                   feedbackViewController.view.isDescendant(of: window) {
+                    feedbackViewController.view.removeFromSuperview()
+                    window.addSubview(feedbackViewController.view)
+                }
+            }
```

Comment:
Oh right, this is our glorious hack for getting style changes to work with UIAppearance. It would be kind of embarrassing to expect developers to write this code on their own and even put it in an example. Maybe we should expose `StyleManager.forceRefreshAppearanceIfNeeded()` publicly somehow? Or is there a way to integrate the feedback view controller with the existing `StyleManagerDelegate.styleManager(_:viewForApplying:)` and `StyleManagerDelegate.styleManagerDidRefreshAppearance(_:)` methods?

Reply:
https://github.com/mapbox/mapbox-navigation-ios/pull/2897 has brought the specific current viewController's view to refresh, and excluded other views such as the prompt up presented view like the feedback window. If we expose the `StyleManager.forceRefreshAppearanceIfNeeded()` and use the `StyleManagerDelegate.styleManager(_:viewForApplying:)` , the `StyleManager .forceRefreshAppearance()` would apply both the current viewController's view and the desired view to refresh.  It would cause significant style change delay. For example, if we applied the `presentedViewController`  view in the free driving, and do the same in the active navigation. We could see that both of them would be called, so the NavigationView would have significant delay because of that.

Maybe we should revert https://github.com/mapbox/mapbox-navigation-ios/pull/2897 , apply style change to all views and add the `window.isKind(of: NSClassFromString("UITextEffectsWindow") ?? NSString.classForCoder())` check to avoid the https://github.com/mapbox/mapbox-navigation-ios/issues/2764 issue.

Reply:
Reverting #2897 would have some unfortunate consequences for applications that present keyboards. Let’s ticket out some tail work to streamline applying this update. Perhaps we could encapsulate this code in a utility method, but we should give it some thought before formally adding it to the public API. In the meantime, customers who really need this fix can copy this part of the example application, with the understanding that it’ll get easier in the future.

Reply:
`StyleManagerDelegate.styleManager(_:viewForApplying:)` has been kept and marked as deprecated, but removed from the `StyleManager.forceRefreshAppearance()`. This code only works for the `presentedViewController` already in the window, and is currently included in the `StyleManager.forceRefreshAppearance()` because it updates all the windows view except the key board one. If customer has their own view to update, it should be included in the `StyleManager.forceRefreshAppearance()` already without the specifying.